CMakeLists.txt

```diff
@@ -418,7 +418,7 @@ if(ARB_WITH_GPU)
         target_compile_options(arbor-private-deps INTERFACE $<$<COMPILE_LANGUAGE:CXX>:${clang_options_}>)
         target_compile_options(arborenv-private-deps INTERFACE $<$<COMPILE_LANGUAGE:CXX>:${clang_options_}>)
     elseif(ARB_WITH_HIP_CLANG)
-        set(clang_options_ -DARB_HIP -xhip --amdgpu-target=gfx906 --amdgpu-target=gfx900)
+        set(clang_options_ -DARB_HIP -xhip --offload-arch=gfx906 --offload-arch=gfx900)
```

Comment:
This has been a þ in my side for quite some time, but never came round to fix it. Can we detect/generalise or consume user args, similar to `CMAKE_CUDA_ARCHITECTURES`? If not it's not a big deal.